record/record.controller.js

```diff
@@ -116,52 +120,71 @@
         };
 
         /**
+         * {boolean} outputWVersion - true if version should be included in output
          * The following cases need to be handled for the resolverImplicitCatalog value:
-         *  - if undefined:                                 use current chaise path
+         *  - if null:                                      use current chaise path
          *  - if false:                                     /id/currCatalog/RID
+         *  - if undefined:                                 same as false
          *  - if resolverImplicitCatalog == currCatalog:    /id/RID
          *  - if resolverImplicitCatalog != currCatalog:    /id/resolverImplicitCatalog/RID
          **/
-        var resolvePermalink = function (tuple) {
+        var resolvePermalink = function (tuple, outputWVersion) {
```

Comment:
This is very difficult to read and I feel like it's going to be a pain to maintain. Also I don't think you can use version number of a catalog with another one. can you? In the case that data-modelers have defined a `resolverId` that is different from the current catalog, you're adding the version of current catalog.

What's the purpose of the new input? To always return a version? can we do it in all these cases? It might be easier to avoid mixing it with the current way this function works.

Reply:
> I don't think you can use version number of a catalog with another one. can you?

You are never getting put into another catalog. The resolver config tells us which catalog the resolver resolves to by default. If our current catalog id matches the resolver id, then the catalog id doesn't appear in the link. If those ids do not match, we include the current catalog id.

> What's the purpose of the new input? To always return a version? can we do it in all these cases?

This is so that the function can return both the live link and the versioned link. `outputWVersion` is a boolean that controls if a link will include a version or not. This function will have to change though for the case that was discussed during scrum yesterday morning. There are 4 cases that can occur that the function will output for:
 1. no version in url
    - output live link
    - output versioned link for live (next issue I'm working on)
       - this case is ignored by the function right now because of a conditional in the open modal function
       - this is the reason I need to refactor
 2. version in url
    - output live link
    - output current versioned link

Reply:
So `outputWVersion` will only work when the resolver is going to use the current catalog id, or no catalog id. It might be easier to not mix it with the other cases. Especially the `rewriteUrlWithoutVersion` function that despite its name, will return the version in some cases.